packages/back-end/test/services/experiments.test.ts

```diff
@@ -0,0 +1,310 @@
+      );
+    });
+  });
+
+  describe("postMetricApiPayloadToMetricInterface", () => {
+    describe("SQL datasource", () => {
+      const datasource: Pick<DataSourceInterface, "type"> = {
+        type: "postgres",
+      };
+      const organization: OrganizationInterface = {
+        id: "org_abc123",
+        url: "",
+        dateCreated: new Date(),
+        name: "Acme Donuts",
+        ownerEmail: "acme@acme-donuts.net",
+        members: [],
+        invites: [],
+      };
+
+      it("with minimum payload, should create a MetricInterface from a postMetric payload", () => {
+        const input: z.infer<typeof postMetricValidator.bodySchema> = {
+          datasourceId: "ds_abc123",
+          sql: {
+            identifierTypes: ["user_id"],
+            conversionSQL: "select * from foo",
+            userAggregationSQL: "sum(values)",
+          },
+          name: "My Cool Metric",
+          type: "binomial",
+        };
+
+        const result = postMetricApiPayloadToMetricInterface(
+          input,
+          organization,
+          datasource
+        );
+
+        expect(result.aggregation).toEqual("sum(values)");
+        expect(result.conditions).toEqual([]);
+        expect(result.datasource).toEqual("ds_abc123");
+        expect(result.denominator).toBe(undefined);
+        expect(result.description).toEqual("");
+        expect(result.ignoreNulls).toEqual(false);
+        expect(result.inverse).toEqual(false);
+        expect(result.name).toEqual("My Cool Metric");
+        expect(result.organization).toEqual("org_abc123");
+        expect(result.owner).toEqual("");
+        expect(result.projects).toEqual([]);
+        expect(result.tags).toEqual([]);
+        expect(result.queries).toEqual([]);
+        expect(result.queryFormat).toEqual("sql");
+        expect(result.runStarted).toEqual(null);
+        expect(result.sql).toEqual("select * from foo");
+        expect(result.type).toEqual("binomial");
+        expect(result.userIdTypes).toEqual(["user_id"]);
+      });
+
+      it("with a full payload, should create a MetricInterface from a postMetric payload", () => {
+        const input: z.infer<typeof postMetricValidator.bodySchema> = {
+          datasourceId: "ds_abc123",
+          tags: ["checkout"],
+          projects: ["proj_abc987"],
+          sqlBuilder: {
+            tableName: "users",
+            timestampColumnName: "created_at",
+            valueColumnName: "signed_up",
+            conditions: [
+              {
+                value: "true",
+                operator: "=",
+                column: "signed_up",
+              },
+            ],
+            identifierTypeColumns: [
+              {
+                columnName: "id",
+                identifierType: "string",
+              },
+            ],
+          },
+          behavior: {
+            goal: "decrease",
+            conversionWindowStart: 10,
+            conversionWindowEnd: 50,
```

Comment:
These result in:

```ts
        expect(result.conversionWindowHours).toEqual(40);
        expect(result.conversionDelayHours).toEqual(10);
```

From my understanding, this calculation is correct. Let me know if it isn't.

Reply:
Yes, this is correct.